src/balancecommand.cpp

```diff
@@ -0,0 +1,180 @@
+/****************************************************************************************
+**
+** Copyright (C) 2015 Jolla Ltd.
+** Contact: Martin Grimme <martin.grimme@gmail.com>
+** All rights reserved.
+**
+** This file is part of the btrfs-balancer package.
+**
+** You may use this file under the terms of the GNU Lesser General
+** Public License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+** Lesser General Public License for more details.
+**
+****************************************************************************************/
+
+
+#include "balancecommand.h"
+#include "btrfsbalancer.h"
+#include "dbusservice.h"
+
+#include <QDBusConnection>
+#include <QDBusInterface>
+#include <QDBusMessage>
+#include <QFile>
+
+#include <contextproperty.h>
+
+#include <iostream>
+
+namespace
+{
+const QString BATTERY_CHARGE("Battery.ChargePercentage");
+const QString BATTERY_STATUS("Battery.IsCharging");
+
+int batteryCharge()
+{
+    return ContextProperty(BATTERY_CHARGE).value(QVariant(0)).toInt();
```

Comment:
Battery.State is marked as deprecated on https://autodoc.jollamobile.com/statefs-providers/latest-devel/group__statefs__properties.html

I'm gonna use it anyway, since you said so, and it seems to be the only way to catch the "unkown" state.